patterns/observer/observer.js

```diff
@@ -0,0 +1,49 @@
+// Observer pattern defines a one to many dependency relationship from Subject to Observers
+// Observers wait for a signal / trigger from the subject to start.
+// Code reference: https://www.youtube.com/watch?v=PldXoGemkyk
+
+
+// OBSERVER PATTERN WITH CLASS
+
+
+
+class Observable {
+
+    constructor() {
+
+        // The list of subscribers
+        this.subscribers = []
+    }
+
+    // Function is added to the list of subscribers
+    subscribe(fn) {
+        this.subscribers.push(fn);
+    }
+
+    // Function will get removed from the subscribers list
+    unsubscribe(fn) {
+        this.subscribers = this.subscribers.filter((item) => item !== fn);
+    }
+
+    // To broadcast to all the observers
+    broadcast(data) {
+        for (let i = 0; i < this.subscribers.length; i++) {
+            // Calling each function in the observers list with the broadcast data
+            this.subscribers[i](data);
+        }
+    }
+}
+
+
+const observer = new Observable();
+
+const fn = (data) => {
+    console.log("Callback was executed with data", data);
+};
+
+// function 'fn' subscribed to Observable
+observer.subscribe(fn);
+
+
+// Broadcasting message to all the subscribers
+observer.broadcast("Hello from observable");
```

Comment:
The problem discussed above is clear here from the semantics of this line: Line 49 says that the _observer_ is broadcasting information; in fact, according to the [description of the pattern](https://en.wikipedia.org/wiki/Observer_pattern), the _subject_ should be broadcasting (i.e. notifying) all _observers_ (in this case, your `const fn`).